### ISSUE TITLE
Add confirmation whem close Foundry.

### ISSUE THREAD

**Issue by gandb** [foundryvtt/foundryvtt#10981]
### User Experience

When I will close one character sheet is very common close the server. Please, put a confirmation before close.

**Comment by aaclayton** [comment]
I'm not sure it's possible to intercept or override the behavior of the windows application close button, but we can look into whether that may be possible.

**Comment by gandb** [comment]
Yes, it is possible. I will create a PR, but first I want to know if the team is interested in this change

**Comment by aaclayton** [comment]
We only keep issues open in this backlog if we are interested in exploring them at some point, so yes. You won't be able to create a PR though because our source code does not exist in a public repo. Feel free to post your suggested approach here!

**Comment by gandb** [comment]
Can see if this help you? 
```

const { app, BrowserWindow } = require('electron')

app.once('ready', () => {
  let win = new BrowserWindow()
  let isRegularClose = false
  setTimeout(() => {
    if (win) win.destroy()
  }, 5000)
  win.on('close', (event) => {
    isRegularClose = true
  })
  win.on('closed', (event) => {
    console.log(isRegularClose
      ? 'win closed (X)'
      : 'win destroyed (code)')
    win = null
  })
})
```
Source [https://stackoverflow.com/questions/48408014/electron-catch-when-window-is-closed-using-the-x-button](https://stackoverflow.com/questions/48408014/electron-catch-when-window-is-closed-using-the-x-button)

**Comment by aaclayton** [comment]
These events - as far as I am aware - do not allow you to prevent the window from closing. They allow you to react with a short synchronous method, but you cannot stop the window from closing.

**Comment by aaclayton** [comment]
Yes, `dialog.showMessageBoxSync` might work for this. Thanks I'll look into it.
